src/Chapter3.hs

```diff
@@ -910,6 +1003,21 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = Gold Int
+
+instance Append Gold where
+    append :: Gold -> Gold -> Gold
+    append (Gold a) (Gold b) = Gold (a + b)
+
+instance Append [a] where
+    append :: [a] -> [a] -> [a]
+    append a b = a ++ b
+
+instance (Append a) => Append (Maybe a) where
+    append :: (Append a) => Maybe a -> Maybe a -> Maybe a
+    append Nothing (Just a) = Just a
+    append (Just a) Nothing = Just a
+    append (Just a) (Just b) = Just (append a b)
```

Comment:
This function is partial and doesn't cover the case when both `Maybe`s are `Nothing`. If you build the project with the `make build` command, you will see warnings that patterns are not exhaustive here.